### DISCUSSION TITLE
Bug report: the tool suggests to write the same multiple times

### DISCUSSION THREAD

**Discussion by dudifrid** [community/community#7387] [Copilot Conversations]
When I place the cursor BEFORE a line of code that has been suggested by the tool, it suggests the same again, even though it can easily figure out that it is already written a few lines below, so there is no reason to suggest to write this again.

This usually happens when I place the cursor one line before a snippet of code that has been suggested by the tool (hereinafter, "that snippet of code"), but it can also happen when I place the cursor a few lines away from that snippet of code, and even if there are several of lines of code betweeen the cursor and that snippet of code.

**Comment by MulleDK19** [comment]
That's just a side effect of how it works. It doesn't know about anything that is after the caret.

**Reply by dudifrid** [reply]
Yet it is a bug, and in my opinion should be fixed.
I find it rather annoying, and I believe that it must read the next few lines, to receive better results.

**Reply by MulleDK19** [reply]
It's not, and it's definitely not annoying. If I go back to a point inside a method and it read what's after, it would generate completely incorrect code. If you need to go back, write code or comment to guide it.

The only time where this is a problem is separate methods where methods after the one you're working in is unbeknownst.
